src/plugins/data/public/ui/dataset_selector/dataset_table.tsx

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiBasicTable, EuiFieldSearch, EuiLink, EuiText } from '@elastic/eui';
+import React, { useRef, useState } from 'react';
+import { FormattedMessage } from '@osd/i18n/react';
+import { DataStructure } from '../../../common';
+import { DataStructureFetchOptions } from '../../query/query_string/dataset_service';
+import { getQueryService } from '../../services';
+
+interface DatasetTableProps {
+  path: DataStructure[];
+  setPath: (newPath: DataStructure[]) => void;
+  index: number;
+  explorerDataset: DataStructure | undefined;
+  selectDataStructure: (item: DataStructure | undefined, newPath: DataStructure[]) => Promise<void>;
+  fetchNextDataStructure: (
+    nextPath: DataStructure[],
+    dataType: string,
+    options?: DataStructureFetchOptions
+  ) => Promise<DataStructure>;
+}
+
+export const DataSetTable: React.FC<DatasetTableProps> = (props) => {
+  const datasetService = getQueryService().queryString.getDatasetService();
+  const [loading, setLoading] = useState(false);
+  const searchRef = useRef<HTMLInputElement | null>(null);
+
+  const initialSelectedIds = props.explorerDataset?.id.split(',');
+  const dataStructures = props.path[props.index].children || [];
+  const nextToken = props.path[props.index].nextToken;
+
+  const onTableChange = async (options: DataStructureFetchOptions) => {
+    const typeConfig = datasetService.getType(props.path[1].id);
+    if (!typeConfig) return;
+
+    setLoading(true);
+    await props
+      .fetchNextDataStructure(props.path, typeConfig.id, options)
+      .then((newDataStructure) => {
+        props.setPath([...props.path.slice(0, props.index), newDataStructure]);
+      })
+      .finally(() => setLoading(false));
+  };
+
+  return (
+    <div className="datasetTable">
+      <EuiFieldSearch
+        fullWidth
+        inputRef={(node) => (searchRef.current = node)}
+        onSearch={(value) => onTableChange({ search: value })}
+      />
+
+      <EuiBasicTable
+        items={dataStructures}
+        itemId="id"
+        columns={[{ field: 'title', name: 'Name' }]}
+        loading={loading}
+        isSelectable
+        selection={{
+          onSelectionChange: (items) => {
+            if (items.length === 0) {
+              props.selectDataStructure(undefined, props.path.slice(0, props.index + 1));
+              return;
+            }
+            if (!items.every((item) => item.type === items[0].type)) {
+              throw new Error('All items must be of the same type');
+            }
+            const newItem: DataStructure = {
```

Comment:
what if we made multidatasets apart of the dataset service so if another plugin would want to implement a solution instead of using our selector they could.

so the [cacheDataset](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/8d52134734857561ad580d2567c68c01fe3e9fa8/src/plugins/data/public/query/query_string/dataset_service/dataset_service.ts#L65) has the ability to take in an array of data structures. 

for example, the dataset type config for example S3 tables will let the dataset service so we dont need to worry about creating some data structure on the fly but just keeping an array of data structures in the ui portion. then when the user selected the data structures and hit next that is when we construct the dataset properly joining the ids.

this way then we dont have to worry about slicing and joining ID before they are ready. also lets plugin developers use the dataset service without using the data plugin ui

Reply:
do you mean changing `selectedDatasets` to an array in advanced selector https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1750cadee4d94c62532ddff377c74180c2b19ceb/src/plugins/data/public/ui/dataset_selector/advanced_selector.tsx#L51

when user clicks next, configurator receives the list of datasets and uses dataset service to combine them into a single dataset?

Reply:
actually it would change the `path` definition, might be too much change https://github.com/opensearch-project/OpenSearch-Dashboards/blob/f027944f67cf2c6a6439fb3b6c271920f4d60efc/src/plugins/data/public/ui/dataset_selector/dataset_explorer.tsx#L52-L73
if we defer combining ids until `next`, we need to properly represent `path` to include the multiple selected datasets in `dataset_explorer`. but `path` is `DataStructure[]`, we can't put multiple datasets on the same level. or did i understand your suggestion correctly?

Reply:
i see what you mean. this MVP to get multi select working without being too intrusive of a change. i can stand by that. should we create a backlog item to revisit this to make the dataset service to the heavy lifting of combining data structures

Reply:
https://github.com/opensearch-project/OpenSearch-Dashboards/issues/8362